### PR TITLE
fix: tolerate invalid RegExp patterns in archive search

### PR DESCRIPTION
**Pull request by aleclarson**
To reproduce:
- Unlock an archive
- Use cmd+F
- Type "s["

__Buttercup v1.12.2__

### CODE REVIEW

File: src/renderer/components/archive/archive-search.js [hunk 1]
```diff
@@ -177,8 +177,11 @@ class ArchiveSearch extends PureComponent {
    * @memberof ArchiveSearch
    */
   highlightSearchResult = word => {
-    const regex = new RegExp('(' + this.state.searchTerm + ')', 'ig');
-    return word.replace(regex, '<mark>$1</mark>');
+    try {
+      const regex = new RegExp('(' + this.state.searchTerm + ')', 'ig');
+      return word.replace(regex, '<mark>$1</mark>');
+    } catch(e) {}
+    return word;
```
Comment (perry-mitchell) [inline src/renderer/components/archive/archive-search.js]: Good idea! Though I'd recommend making this simpler now that we're looking at it. Using a regex is obviously going to cause issues when special chars are used, so we should opt for plain text.

Simply finding the index and replacing the word from that point only should do the trick What about something like this:

```javascript
const index = word.toLowerCase().indexOf(this.state.searchTerm.toLowerCase());
if (index >= 0) {
  return `${word.substring(0, index)}<mark>${this.state.searchTerm}</mark>${word.substring(index + this.state.searchTerm.length)}`;
}
return word;
```

Comment (perry-mitchell) [reply src/renderer/components/archive/archive-search.js]: I haven't tested if this works, but it'd do away with the complexity of regular expressions.

Comment (aleclarson) [reply src/renderer/components/archive/archive-search.js]: How about escaping any `RegExp` characters in the search term using [escape-string-regexp](https://www.npmjs.com/package/escape-string-regexp)?

Comment (aleclarson) [reply src/renderer/components/archive/archive-search.js]: We can avoid the dependency by copy-pasting.

```js

  highlightSearchResult = word => {
    const searchTerm = this.state.searchTerm.replace(/[|\\{}()[\]^$+*?.]/g, '\\$&');
    const regex = new RegExp('(' + searchTerm + ')', 'ig');
    return word.replace(regex, '<mark>$1</mark>');
  };
```

Comment (aleclarson) [reply src/renderer/components/archive/archive-search.js]: Pushed. 😄

Comment (perry-mitchell) [reply src/renderer/components/archive/archive-search.js]: Sorry I didn't reply sooner. It's still not as robust as what I suggested, I'm afraid. There's no need for regex at all in this situation. It's a simple result of highlighting the first instance of the search term, which my example does with only string splitting.

**Comment by sallar** [comment]
@perry-mitchell this looks good to me, are you happy with it too mate?

**Pull request merged**